src/libstd/sync/rwlock.rs

```diff
@@ -330,6 +330,19 @@ impl<T: ?Sized + fmt::Debug> fmt::Debug for RwLock<T> {
     }
 }
 
+#[stable(feature = "std_guard_impls", since = "1.20")]
+impl<T: ?Sized + fmt::Display> fmt::Display for RwLock<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self.try_read() {
+            Ok(guard) => write!(f, "RwLock {{ data: {} }}", guard),
```

Comment:
I'm not 100% satisfied with this representation for `Display`, I'm just not sure how it should be given that we could be unable to effectively lock the guard and get at the internal data